agent/lib_aws_sdk_php.c

```diff
@@ -519,6 +518,171 @@ char* nr_lib_aws_sdk_php_get_command_arg_value(char* command_arg_name,
+      && NULL != cloud_attrs->cloud_region) {
+    /* Must be freed by caller */
+    cloud_attrs->cloud_resource_id = nr_formatf(
+        "arn:aws:dynamodb:%s:%s:table/%s", cloud_attrs->cloud_region,
+        account_id, datastore_params->collection);
+  }
+}
+
+void nr_lib_aws_sdk_php_dynamodb_handle(nr_segment_t* auto_segment,
+                                        char* command_name_string,
+                                        size_t command_name_len,
+                                        NR_EXECUTE_PROTO) {
+  nr_segment_t* datastore_segment = NULL;
+  nr_segment_cloud_attrs_t cloud_attrs = {0};
+  nr_datastore_instance_t instance = {0};
+  nr_segment_datastore_params_t datastore_params = {
+    .db_system = AWS_SDK_PHP_DYNAMODBCLIENT_DATASTORE_SYSTEM,
+    .datastore = {
+      .type = NR_DATASTORE_DYNAMODB,
+    },
+    .instance = &instance,
+    .callbacks = {
+      .backtrace = nr_php_backtrace_callback,
+    },
+  };
+  if (NULL == auto_segment) {
+    return;
+  }
+
+  if (NULL == command_name_string || 0 == command_name_len) {
+    return;
+  }
+
+#define AWS_COMMAND_IS(CMD) \
+  (command_name_len == (sizeof(CMD) - 1) && nr_streq(CMD, command_name_string))
+
+  /* Determine if we instrument this command. */
+  if (AWS_COMMAND_IS("createTable")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_CREATE_TABLE;
+  } else if (AWS_COMMAND_IS("deleteItem")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_DELETE_ITEM;
+  } else if (AWS_COMMAND_IS("deleteTable")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_DELETE_TABLE;
+  } else if (AWS_COMMAND_IS("getItem")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_GET_ITEM;
+  } else if (AWS_COMMAND_IS("putItem")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_PUT_ITEM;
+  } else if (AWS_COMMAND_IS("query")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_QUERY;
+  } else if (AWS_COMMAND_IS("scan")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_SCAN;
+  } else if (AWS_COMMAND_IS("updateItem")) {
+    datastore_params.operation = AWS_SDK_PHP_DYNAMODBCLIENT_UPDATE_ITEM;
+  } else {
+    /* Nothing to do here so exit. */
+    return;
+  }
+#undef AWS_COMMAND_IS
+
+  /*
+   * nr_lib_aws_sdk_php_dynamodb_set_params sets:
+   * the cloud_attrs->region and cloud_resource_id(needs to be freed)
+   * datastore->instance host and port_path_or_id(needs to be freed)
+   * datastore->collection (needs to be freed)
+   */
+  nr_lib_aws_sdk_php_dynamodb_set_params(&datastore_params, &cloud_attrs,
+                                         NR_EXECUTE_ORIG_ARGS);
+
+  /*
+   * By this point, the datastore params are decoded, grab the parent segment
+   * start time, add the special segment attributes/metrics then close the newly
+   * created segment.
+   */
+  datastore_segment = nr_segment_start(NRPRG(txn), NULL, NULL);
+  if (NULL == datastore_segment) {
+    return;
```

Comment:
Will the parameters mentioned in the comment just above going to be freed properly after this `return`?

Reply:
Good catch, please see: [02c00c9](https://github.com/newrelic/newrelic-php-agent/pull/1037/commits/02c00c96d55cbddc1e8b71c2cfcc3b56f343bba7)